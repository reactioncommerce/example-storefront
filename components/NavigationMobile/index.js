export { default as NavigationMobile } from "./NavigationMobile";
export { default as NavigationItemMobile } from "./NavigationItemMobile";
export { default as NavigationToggleMobile } from "./NavigationToggleMobile";
