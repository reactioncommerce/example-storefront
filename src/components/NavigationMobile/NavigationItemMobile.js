import React, { Component, Fragment } from "react";
import PropTypes from "prop-types";
import { inject } from "mobx-react";
import { Router } from "routes";
import Divider from "@material-ui/core/Divider";
import ListItemIcon from "@material-ui/core/ListItemIcon";
import ListItemText from "@material-ui/core/ListItemText";
import Collapse from "@material-ui/core/Collapse";
import MenuList from "@material-ui/core/MenuList";
import MenuItem from "@material-ui/core/MenuItem";
import ChevronDownIcon from "mdi-material-ui/ChevronDown";
import ChevronUpIcon from "mdi-material-ui/ChevronUp";
import { withStyles } from "@material-ui/core/styles";

const styles = (theme) => theme.NavigationItemMobile;

@inject("routingStore")
@withStyles(styles)
class NavigationItemMobile extends Component {
  static propTypes = {
    classes: PropTypes.object,
    navItem: PropTypes.object,
    routingStore: PropTypes.object
  };

  static defaultProps = {
    classes: {},
    navItem: {},
    routingStore: {}
  };

  state = { isSubNavOpen: false };

  get linkPath() {
    const { navItem, routingStore } = this.props;
    return routingStore.queryString !== ""
      ? `/tag/${navItem.slug}?${routingStore.queryString}`
      : `/tag/${navItem.slug}`;
  }

  get hasSubNavItems() {
    const { navItem: { subTags } } = this.props;
    return Array.isArray(subTags.edges) && subTags.edges.length > 0;
  }

  onClick = () => {
    const { navItem } = this.props;

    if (this.hasSubNavItems) {
      this.setState({ isSubNavOpen: !this.state.isSubNavOpen });
    } else {
      const path = this.linkPath;
      Router.pushRoute(path, { slug: navItem.slug });
    }
  };

  onClose = () => {
    this.setState({ isSubNavOpen: false });
  };

  renderSubNav(navItemGroup) {
    const { classes } = this.props;
    return (
      <div className={classes.subNav}>
        <Divider />
        {navItemGroup.subTags.edges.map(({ node: navItemGroupItem }, index) => (
          <MenuItem className={classes.nested} dense inset key={index}>
            <ListItemText classes={{ inset: classes.listItemTextInset }} inset primary={navItemGroupItem.name} />
          </MenuItem>
        ))}
      </div>
    );
  }

  renderCollapse() {
    const { classes, navItem: { subTags } } = this.props;
    return (
      <Collapse in={this.state.isSubNavOpen} timeout="auto" unmountOnExit>
        <MenuList component="div" disablePadding>
          {subTags.edges.map(({ node: navItemGroup }, index) => (
            <MenuList disablePadding key={index}>
              <MenuItem inset className={classes.nested}>
                <ListItemText classes={{ inset: classes.listItemTextInset }} inset primary={navItemGroup.name} />
              </MenuItem>
              {Array.isArray(navItemGroup.subTags.edges) && this.renderSubNav(navItemGroup)}
            </MenuList>
          ))}
        </MenuList>
      </Collapse>
    );
  }

  render() {
    const { classes, navItem } = this.props;
    return (
      <Fragment>
        <MenuItem color="inherit" onClick={this.onClick}>
          <ListItemText classes={{ primary: classes.primary }} primary={navItem.name} />
          {this.hasSubNavItems && (
            <ListItemIcon className={classes.icon}>
              {this.state.isSubNavOpen ? <ChevronUpIcon /> : <ChevronDownIcon />}
            </ListItemIcon>
          )}
        </MenuItem>
        {this.hasSubNavItems && this.renderCollapse()}
      </Fragment>
    );
  }
}

export default NavigationItemMobile;
