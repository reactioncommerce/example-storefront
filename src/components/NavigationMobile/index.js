export { default as NavigationMobile } from "./NavigationMobile";
