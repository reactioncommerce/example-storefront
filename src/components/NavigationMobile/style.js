import styled from "styled-components";

export const Container = styled.div`
  flex: 0 0 auto;
  font-family: ${({ theme }) => theme.typography.fontFamily};

  button {
    display: flex;
    align-items: center;
    width: 100%;
    background: transparent;
    color: #4f4d4d;
    border: none;
    user-select: none;
    outline: none;
  }
`;

export const Header = styled.div`
  padding: 0px 30px;
`;

export const Footer = styled.div`
  margin-top: 30px;
`;

export const LinkContainer = styled.div`
  margin-bottom: 5px;
  padding: 0px;

  a {
    display: flex;
    align-items: center;
    width: 100%;
    height: 41px;
    font-size: 16px;
    outline: none;
    padding: 0px 30px;
    color: "blue";
    font-size: 14px;

    svg {
      margin-left: -5px;
      margin-right: 20px;
    }
  }
`;
