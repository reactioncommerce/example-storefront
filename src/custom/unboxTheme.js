const theme = {
  typography: {
    fontFamily: "Raleway",
    colors: {
      primary: "#FFF",
      secondary: "#4f4d4d",
      tertiary: "#9c27b1",
      price: "#00c3cb"
    },
    sizes: {
      title: {
        SIZE_1: "35px",
        SIZE_2: "30px",
        SIZE_3: "25px",
        SIZE_4: "20px"
      },
      text: {
        SIZE_1: "18px",
        SIZE_2: "16px",
        SIZE_3: "14px",
        SIZE_4: "12px",
        SIZE_5: "10px"
      }
    },
    weights: {
      NORMAL: 400,
      SEMI: 600,
      BOLD: 700
    }
  },
  layout: {
    backgrounds: {
      primary: "#FFF",
      secondary: "#f6f6f6"
    },
    colors: {
      primary: "#FFF",
      secondary: "#00c3cb",
      tertiary: "#d8f5f7"
    }
  }
};

export default theme;
