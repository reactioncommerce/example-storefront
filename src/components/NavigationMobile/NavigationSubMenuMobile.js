import React, { Component } from "react";
import PropTypes from "prop-types";
import Divider from "@material-ui/core/Divider";
import Toolbar from "@material-ui/core/Toolbar";
import IconButton from "@material-ui/core/IconButton";
import Typography from "@material-ui/core/Typography";
import MenuList from "@material-ui/core/MenuList";
import ChevronLeftIcon from "mdi-material-ui/ChevronLeft";
import { withStyles } from "@material-ui/core/styles";
import NavigationItemMobile from "./NavigationItemMobile";

const styles = () => ({
  root: {
    display: "flex",
    flexDirection: "column",
    height: "100vh"
  },
  header: {
    flex: "0 0 auto"
  },
  toolbarTitle: {
    position: "absolute",
    width: "100%",
    textAlign: "center"
  },
  menu: {
    flex: "1 1 auto",
    overflowY: "auto"
  }
});

class NavigationSubMenuMobile extends Component {
  static propTypes = {
    classes: PropTypes.object,
    navItem: PropTypes.object,
    onBackButtonClick: PropTypes.func,
    routingStore: PropTypes.object
  };

  static defaultProps = {
    classes: {},
    navItem: {},
    onBackButtonClick() {},
    routingStore: {}
  };

  state = { isSubNavOpen: false };

  get hasSubNavItems() {
    const { navItem: { items } } = this.props;
    return Array.isArray(items) && items.length > 0;
  }

  render() {
    const { classes, navItem } = this.props;

    if (!navItem) {
      return null;
    }

    return (
      <div className={classes.root}>
        <div className={classes.header}>
          <Toolbar disableGutters>
            <Typography className={classes.toolbarTitle} variant="subtitle1">{navItem.navigationItem.data.contentForLanguage}</Typography>
            <IconButton onClick={this.props.onBackButtonClick}>
              <ChevronLeftIcon />
            </IconButton>
          </Toolbar>
          <Divider />
        </div>
        <div className={classes.menu}>
          {this.hasSubNavItems &&
            <MenuList component="div" disablePadding>
              {navItem.items.map((item, index) => (
                <NavigationItemMobile key={index} navItem={item} />
              ))}
            </MenuList>
          }
        </div>
      </div>
    );
  }
}

export default withStyles(styles)(NavigationSubMenuMobile);