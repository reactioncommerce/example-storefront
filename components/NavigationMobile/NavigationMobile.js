import React, { Component } from "react";
import PropTypes from "prop-types";
import { inject, observer } from "mobx-react";
import ShopLogo from "@reactioncommerce/components/ShopLogo/v1";
import { withStyles } from "@material-ui/core/styles";
import Divider from "@material-ui/core/Divider";
import Drawer from "@material-ui/core/Drawer";
import IconButton from "@material-ui/core/IconButton";
import MenuList from "@material-ui/core/MenuList";
import Slide from "@material-ui/core/Slide";
import Toolbar from "@material-ui/core/Toolbar";
import Typography from "@material-ui/core/Typography";
import CloseIcon from "mdi-material-ui/Close";
import Link from "components/Link";
import NavigationItemMobile from "./NavigationItemMobile";
import NavigationSubMenuMobile from "./NavigationSubMenuMobile";

const styles = (theme) => ({
  root: {
    display: "flex",
    flexDirection: "column"
  },
  header: {
    flex: "0 0 auto"
  },
  toolbarTitle: {
    position: "absolute",
    width: "100%",
    textAlign: "center"
  },
  title: {
    display: "inline-block",
    color: theme.palette.reaction.reactionBlue,
    borderBottom: `solid 5px ${theme.palette.reaction.reactionBlue200}`
  },
  menu: {
    flex: "1 1 auto",
    overflowY: "auto",
    width: 320
  },
  subNav: {
    position: "absolute",
    top: 0,
    left: 0,
    width: 320,
    height: "100vh",
    backgroundColor: theme.palette.background.default
  }
});

class NavigationMobile extends Component {
  static propTypes = {
    classes: PropTypes.object,
    navItems: PropTypes.object,
    shop: PropTypes.shape({
      name: PropTypes.string
    }),
    uiStore: PropTypes.shape({
      closeMenuDrawer: PropTypes.func
    }).isRequired
  };

  static defaultProps = {
    classes: {},
    navItems: []
  };

  state = {
    navItem: null
  }

  handleNavItemClick = (navItem) => {
    this.setState({
      navItem
    });
  }

  handleCloseSubMenu = () => {
    this.setState({ navItem: null });
  }

  renderNavItem = (navItem, index) => (
    <NavigationItemMobile
      key={index}
      isTopLevel
      navItem={navItem}
      onClick={this.handleNavItemClick}
    />
  );

  handleClose = () => {
    this.handleCloseSubMenu();
    this.props.uiStore.closeMenuDrawer();
  };

  render() {
    const { classes, navItems, uiStore, shop } = this.props;

    if (navItems && navItems.items) {
      return (
        <Drawer open={uiStore.isMenuDrawerOpen} onClose={this.handleClose}>
          <div className={classes.header}>
            <Toolbar disableGutters>
              <div className={classes.toolbarTitle}>
                <Typography className={classes.title} color="inherit" variant="h6">
                  <Link route="/" onClick={this.handleClose}>
                    {shop ? <ShopLogo shopName={shop.name} /> : "Example Storefront"}
                  </Link>
                </Typography>
              </div>
              <IconButton onClick={this.handleClose}>
                <CloseIcon />
              </IconButton>
            </Toolbar>
            <Divider />
          </div>
          <nav className={classes.menu}>
            <MenuList disablePadding>{navItems.items.map(this.renderNavItem)}</MenuList>
          </nav>
          <Slide direction="left" in={!!this.state.navItem}>
            <nav className={classes.subNav}>
              <NavigationSubMenuMobile
                navItem={this.state.navItem}
                onBackButtonClick={this.handleCloseSubMenu}
              />
            </nav>
          </Slide>
        </Drawer>
      );
    }

    // If navItems.items aren't available, skip rendering of navigation
    return null;
  }
}

export default withStyles(styles)(inject("navItems", "uiStore")(observer(NavigationMobile)));
