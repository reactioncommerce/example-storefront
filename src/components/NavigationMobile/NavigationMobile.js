import React, { Component } from "react";
import PropTypes from "prop-types";
import { inject, observer } from "mobx-react";
import { withStyles } from "@material-ui/core/styles";
import Drawer from "@material-ui/core/Drawer";
import MenuList from "@material-ui/core/MenuList";
import { NavigationItemMobile } from "components/NavigationMobile";
import withNavigationTags from "../../containers/tags/withNavigationTags";

const styles = (theme) => theme.NavigationMobile;

@withStyles(styles)
@withNavigationTags
@inject("uiStore")
@observer
class NavigationMobile extends Component {
  static propTypes = {
    classes: PropTypes.object,
    navItems: PropTypes.object,
    uiStore: PropTypes.shape({
      closeMenuDrawer: PropTypes.func
    }).isRequired
  };

  static defaultProps = {
    classes: {},
    navItems: []
  };

  renderNavItem(navItem, index) {
    return <NavigationItemMobile key={index} navItem={navItem.node} />;
  }

  handleClose = () => {
    this.props.uiStore.closeMenuDrawer();
  };

  render() {
    const { classes, navItems, uiStore } = this.props;

    return (
      <Drawer open={uiStore.isMenuDrawerOpen} onClose={this.handleClose}>
        <nav className={classes.nav}>
          <MenuList>{navItems.edges && navItems.edges.map(this.renderNavItem)}</MenuList>
        </nav>
      </Drawer>
    );
  }
}

export default NavigationMobile;
