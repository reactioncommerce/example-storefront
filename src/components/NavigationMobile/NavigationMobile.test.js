import React from "react";
import renderer from "react-test-renderer";
import { ThemeProvider } from "styled-components";
import { Provider } from "mobx-react";
import unboxTheme from "../../custom/unboxTheme";
import NavigationMobile from "./NavigationMobile";

const uiStore = {
  isMenuDrawerOpen: false
};

describe("NavigationMobile", () => {
  it("Should match the snapshot", () => {
    const component = renderer.create(
      <ThemeProvider theme={unboxTheme}>
        <Provider uiStore={uiStore}>
          <NavigationMobile />
        </Provider>
      </ThemeProvider>
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
