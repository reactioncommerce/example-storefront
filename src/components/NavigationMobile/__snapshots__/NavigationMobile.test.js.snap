// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationMobile Should match the snapshot 1`] = `null`;
